### PR TITLE
fix: different result from eth_getProof comparing with Ethereum

### PR DESCRIPTION
**Pull request by mmsqe**
Closes: https://github.com/crypto-org-chain/cronos/issues/749

Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: rpc/backend/account_info.go [hunk 1]
```diff
@@ -34,6 +35,23 @@ func (b *Backend) GetCode(address common.Address, blockNrOrHash rpctypes.BlockNu
 	return res.Code, nil
 }
 
+// GetHexProofs returns list of hex data of proof op
+func GetHexProofs(proof *crypto.ProofOps) []string {
```
Comment (fedekunze) [inline rpc/backend/account_info.go:39]: can you move this to utils?

File: rpc/backend/account_info_test.go [hunk 2]
```diff
@@ -373,3 +374,52 @@ func (suite *BackendTestSuite) TestGetTransactionCount() {
...
+		},
+		{
+			"no proof data provided",
+			mookProofs(1, false),
+			defaultRes,
+		},
+		{
+			"valid proof provided",
+			mookProofs(1, true),
+			[]string{"0x040506"},
```
Comment (fedekunze) [inline rpc/backend/account_info_test.go:417]: can we get realistic data here?

File: rpc/backend/account_info_test.go [hunk 3]
```diff
@@ -373,3 +374,52 @@ func (suite *BackendTestSuite) TestGetTransactionCount() {
...
+	if num > 0 {
+		proofOps = &crypto.ProofOps{
+			Ops: make([]crypto.ProofOp, num),
+		}
+		for i := 0; i < num; i++ {
+			proof := crypto.ProofOp{
+				Key: []byte{1, 2, 3},
+			}
+			if withData {
+				proof.Data = []byte{4, 5, 6}
```
Comment (fedekunze) [inline rpc/backend/account_info_test.go:389]: we should use real proofs as test cases instead of 3 bytes

**Review by fedekunze — CHANGES_REQUESTED** [review]
Thanks for the fix! please also add an E2E test besides the unit tests to compare the result with ethereum

**Comment by mmsqe** [comment]
> Thanks for the fix! please also add an E2E test besides the unit tests to compare the result with ethereum

You mean compare like https://github.com/evmos/ethermint/pull/1431/commits/83aa0d0e9fb8cac3bf43f08a570d368dd1ef6949? since I can't do crosscheck, since Merkle proof is a different with ethereum.

**Review by fedekunze — APPROVED** [review]

**Pull request merged**